### PR TITLE
Added filters section and regex, mask filters

### PR DESCRIPTION
**Pull request by baturozgur**
Added filters section to index, added introduction and regex, mask filters.

See in rendered markdown: https://github.com/eddocs/doc/blob/b8e6cf2a131cbca6a0d8d7228f5adb83129011f5/configuration/filters.md

### CODE REVIEW

File: configuration/filters.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  agent, and how the filters are configured for inputs.
+---
+
+# Filters
+
+## Overview
+
+Filters are the mechanism to filter and transform the received logs from input sources.
+
+Proper filter configuration reduces resource usage, increases performance and hides or masks confidential data. It also provides a way to group the related events before further processing.
```
Comment (taylanisikdemir) [inline configuration/filters.md]: how does filters provide a way to group events? are you thinking of buffered trace processor filter? maybe we don't have to mention about that grouping in the overview section.

File: configuration/filters.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Filters apply right after the data ingestion from input and before any processing is done in the workflows associated with the input.
```
Comment (taylanisikdemir) [inline configuration/filters.md]: Filters can be referred at source level, processing rule level and workflow level. let's enrich examples accordingly

File: configuration/filters.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Filters should be defined at the top level, under filters section:
+
+```go
```
Comment (taylanisikdemir) [inline configuration/filters.md]: go is not needed

File: configuration/filters.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  filters:
+    - name: error
+      type: regex
+      pattern: "error"
+```
+Filter names are used to apply filters to inputs.
+
+Below is an example file input with "error" and "mask_card" filters:
```
Comment (taylanisikdemir) [inline configuration/filters.md]: also mention they are applied in given order

File: configuration/filters.md (hunk 1 above)
Comment (baturozgur) [reply configuration/filters.md]: Yes it is buffered tracer.  So I guess we won't plan to mentioned it in the public docs. I am removing that remark.

File: configuration/filters.md (hunk 2 above)
Comment (baturozgur) [reply configuration/filters.md]: Done

File: configuration/filters.md (hunk 4 above)
Comment (baturozgur) [reply configuration/filters.md]: Done

File: configuration/filters.md (hunk 1 above)
Comment (taylanisikdemir) [reply configuration/filters.md]: I will add it to this doc once you merge. but mention about it at the bottom in its own section to leave Overview simple.

**Review by taylanisikdemir — APPROVED** [review]

**Comment by taylanisikdemir** [comment]
Let me know once you merge this, I'll be adding buffered trace filter details

**Pull request merged**